app/models/puppetclass.rb

```diff
@@ -45,30 +46,133 @@ class Puppetclass < ActiveRecord::Base
 
   scope :not_in_any_environment, lambda { includes(:environment_classes).where(:environment_classes => {:environment_id => nil}) }
 
-  # returns a hash containing modules and associated classes
-  def self.classes2hash(classes)
-    hash = {}
-    for klass in classes
-      if (mod = klass.module_name)
-        hash[mod] ||= []
-        hash[mod] << klass
-      else
-        next
+  class << self
```

Comment:
Probably best to keep the refactoring out of this PR if possible, save it for a rubocop related one so this is kept to a minimum, thanks.